### PR TITLE
Fix celldb get

### PR DESCRIPTION
**Pull request by wowtor**
* Parameter `ci` for cell identity is not actually used in `get()` (which renders the cell database useless)

### CODE REVIEW

File: telcell/cell_identity.py [hunk 1]
```diff
@@ -57,16 +57,16 @@ def create(*,
             return NRCell(mcc, mnc, eci)
         elif radio is not None:
             raise ValueError(f"unsupported radio technology: {radio}")
-        elif eci is not None and (ci is None or lac is None):
-            # we have `eci` but not `lac`, `ci` --> guess LTE/NR
+        elif eci is not None:  # we have: `eci`; missing: `radio`
+            if ci is not None or lac is not None:
+                raise ValueError("either `ci` or `eci` should be provided, but not both")
```
Comment (akaIDIOT) [inline telcell/cell_identity.py:62]: The second part of this check is unclear to me: `lac` not being `None`  is not mentioned in the error message?

File: telcell/celldb/pgdatabase.py [hunk 2]
```diff
@@ -41,6 +41,9 @@ def _build_antenna(row: Tuple) -> Properties:
 
 
 def _build_cell_identity_query(ci):
+    if ci is not None and not isinstance(ci, CellIdentity):
```
Comment (akaIDIOT) [inline telcell/celldb/pgdatabase.py:44]: `isinstance` can handle `None`, will be false as well.

```suggestion
    if not isinstance(ci, CellIdentity):
```

File: telcell/celldb/pgdatabase.py [hunk 3]
```diff
@@ -138,11 +141,20 @@ def search(
         count_limit: Optional[int] = 10000,
         random_order: bool = False,
         exclude: Optional[List[CellIdentity]] = None,
+        ci: Optional[CellIdentity] = None,
     ) -> CellCollection:
         qwhere = list(self._qwhere)
         qargs = list(self._qargs)
 
-        if coords is not None and distance_limit_m is not None:
+        if coords is not None and distance_limit_m is None and count_limit is None:
```
Comment (akaIDIOT) [inline telcell/celldb/pgdatabase.py:149]: Are the `None`s here strictly needed? This reads more like English when treated as truthiness imo:

```suggestion
        if coords and not (distance_limit_m or count_limit):
```

File: telcell/celldb/pgdatabase.py [hunk 4]
```diff
@@ -151,21 +163,33 @@ def search(
...
+            qwhere.append(add_qwhere)
+            qargs.extend(add_qargs)
 
-        if mcc is not None:
-            qwhere.append(f"mcc = {mcc}")
-        if mnc is not None:
-            qwhere.append(f"mnc = {mnc}")
+            assert radio is None, "radio argument makes no sense in combination with ci"
+            assert mcc is None, "mcc argument makes no sense in combination with ci"
+            assert mnc is None, "mnc argument makes no sense in combination with ci"
```
Comment (akaIDIOT) [inline telcell/celldb/pgdatabase.py:179]: Is there an intended difference between raising `ValueError`s above and using assertions here? 

Should these be actual errors too? Is the code fine when these assertions are optimized away?

File: telcell/celldb/pgdatabase.py [hunk 5]
```diff
@@ -151,21 +163,33 @@ def search(
...
+        else:
+            if radio is not None:
+                if isinstance(radio, str):
+                    radio = [radio]
+                qwhere.append(f"({' OR '.join(['radio = %s'])})")
+                qargs.extend(radio)
```
Comment (akaIDIOT) [inline telcell/celldb/pgdatabase.py:185]: These seem like sql clauses to be string-formatted into a query later; this makes me very anxious, is there really no other way to do this? :see_no_evil:

**Review by akaIDIOT — APPROVED** [review]

File: telcell/celldb/pgdatabase.py (hunk 5 above)
Comment (wowtor) [reply telcell/celldb/pgdatabase.py:185]: Nope, these are sql (postgres) substitutions, not to be formatted in python

**Pull request merged**